src/raftstore/store/peer_storage.rs

```diff
@@ -308,39 +308,57 @@ impl InvokeContext {
     }
 
     #[inline]
-    pub fn save_raft_to(&self, db: &DB, wb: &mut WriteBatch) -> Result<()> {
-        let handle = try!(rocksdb::get_cf_handle(db, CF_RAFT));
-        try!(wb.put_msg_cf(handle,
-                           &keys::raft_state_key(self.region_id),
-                           &self.raft_state));
+    pub fn save_raft_state_to(&self, wb: &mut WriteBatch) -> Result<()> {
+        try!(wb.put_msg(&keys::raft_state_key(self.region_id), &self.raft_state));
         Ok(())
     }
 
     #[inline]
-    pub fn save_apply_to(&self, db: &DB, wb: &mut WriteBatch) -> Result<()> {
-        let handle = try!(rocksdb::get_cf_handle(db, CF_RAFT));
-        try!(wb.put_msg_cf(handle,
-                           &keys::apply_state_key(self.region_id),
-                           &self.apply_state));
+    pub fn save_apply_state_to(&self, wb: &mut WriteBatch) -> Result<()> {
+        try!(wb.put_msg(&keys::apply_state_key(self.region_id), &self.apply_state));
         Ok(())
     }
 }
 
-fn init_raft_state(engine: &DB, region: &Region) -> Result<RaftLocalState> {
-    Ok(match try!(engine.get_msg_cf(CF_RAFT, &keys::raft_state_key(region.get_id()))) {
+pub fn recover_from_applying_state(engine: &DB, raft_engine: &DB, region_id: u64) -> Result<()> {
+    let raft_wb = WriteBatch::new();
+    box_try!(clear_raft_data(&raft_wb, raft_engine, region_id));
```

Comment:
we can't clear all raft logs here. We may meet a case that in on_raft_ready, we will append some logs after applying the snapshot.